### PR TITLE
Add Smith and Rune Magnus support

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for Smith and Rune Magnus PCBs M75S and M75H.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/smithrune/magnus/m75h/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
+),
+[3] = LAYOUT( /* Base */
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+    KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                   KC_TRNS, KC_TRNS,                   KC_TRNS, KC_TRNS, KC_TRNS
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/smithrune/magnus/m75h/keymaps/default/keymap.c]: Layers removed in [8e8a0d2](https://github.com/qmk/qmk_firmware/pull/22153/commits/8e8a0d2d0e5dd39a1c67981949500daa5d3fb7fe) .

As for the layer layout itself, I have nothing to say really. As much as I agree with you, this is what Smith and Rune wanted as default..

File: keyboards/smithrune/magnus/m75h/info.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "lto": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgblight": true
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/smithrune/magnus/m75h/info.json]: Fixed in [d8fd20c](https://github.com/qmk/qmk_firmware/pull/22153/commits/d8fd20cf4aa816924340e0e01e15b366319c88f2)

File: keyboards/smithrune/magnus/m75s/info.json [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        "backlight": true,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "lto": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgblight": true
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/smithrune/magnus/m75s/info.json]: Fixed in [d8fd20c](https://github.com/qmk/qmk_firmware/pull/22153/commits/d8fd20cf4aa816924340e0e01e15b366319c88f2)

File: keyboards/smithrune/magnus/m75h/m75h.c [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#define CAPS_COLOR_R 0xFF
+#define CAPS_COLOR_G 0xFF
+#define CAPS_COLOR_B 0xFF
+
+void keyboard_post_init_kb(void) {
+    rgblight_set_effect_range(0,29) ;
```
Comment (drashna) [inline keyboards/smithrune/magnus/m75h/m75h.c:23]:
```suggestion
    rgblight_set_effect_range(0,29);
    keyboard_post_init_user();
```

File: keyboards/smithrune/magnus/readme.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Smith and Rune Magnus QMK folder
+
+![magnus](https://smithrune.com/cdn/shop/products/15-SR.jpg)
```
Comment (drashna) [inline keyboards/smithrune/magnus/readme.md:3]: Please rehost this on a dedicated image hosting site (imgur, etc)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/smithrune/magnus/readme.md (hunk 5 above)
Comment (Gondolindrim) [reply keyboards/smithrune/magnus/readme.md:3]: Done in 618decd983

**Comment by Gondolindrim** [comment]
> Sorry, two things I missed earlier.
> 
> The `features` marked `false` in `info.json` are already not enabled by default, so they don't need to be explicitly disabled, but removing these attributes is not a requirement.

I feel it's better to just leave them as `false` if someone needs to enable them later

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/smithrune/magnus/readme.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# Smith and Rune Magnus QMK folder
```
Comment (dunk2k) [inline keyboards/smithrune/magnus/readme.md:1]:
```suggestion
# Smith and Rune Magnus
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Pull request merged**